### PR TITLE
Can filter servicepaths by name

### PR DESCRIPTION
**Pull request by Cerfoglg**
## Proposed changes

A new query option `name` is available to filter servicepaths by a specific name

## Types of changes

What types of changes does your code introduce to the project: _Put
an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after
creating the PR. If you're unsure about any of them, don't hesitate to
ask. We're here to help! This is simply a reminder of what we are going
to look for before merging your code._

- [x] I have read the [CONTRIBUTING][contrib] doc
- [x] I have signed the [Contributor License Agreement][cla]
- [x] I have updated the [RELEASE NOTES][release]
- [x] I have added tests that prove my fix is effective or that my
      feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in
      downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion
by explaining why you chose the solution you did and what alternatives
you considered, etc...

[cla]: https://raw.githubusercontent.com/orchestracities/anubis/master/individual_cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/orchestracities/anubis/blob/master/CONTRIBUTING.md
    "Contributing to Anubis"
[release]: https://github.com/orchestracities/anubis/blob/master/RELEASE_NOTES.md
    "Anubis Release Notes"

### CODE REVIEW

File: anubis-management-api/src/tenants/operations.py [hunk 1]
```diff
@@ -63,10 +63,15 @@ def get_service_paths(db: Session, skip: int = 0, limit: int = 100):
...
+        name: str = None,
         skip: int = 0,
         limit: int = 100):
-    return db.query(models.ServicePath).filter(
-        models.ServicePath.tenant_id == tenant_id).offset(skip).limit(limit).all()
+    db_service_paths = db.query(models.ServicePath).filter(
+        models.ServicePath.tenant_id == tenant_id)
+    if name:
+        db_service_paths = db_service_paths.filter(
+            models.ServicePath.path == name)
```
Comment (chicco785) [inline anubis-management-api/src/tenants/operations.py:73]: should we use a "like" query?

**Review by chicco785 — CHANGES_REQUESTED** [review]
should we use a `like` query to filter by name?

**Comment by Cerfoglg** [comment]
@chicco785 I was about to say. We could have both an option for a "like" query which will match a substring, and a "startswith" query that matches just the start, so one could get just the subpaths for say "/foo", or match any path that happens to contain the word "foo". Thoughts?

**Comment by chicco785** [comment]
> @chicco785 I was about to say. We could have both an option for a "like" query which will match a substring, and a "startswith" query that matches just the start, so one could get just the subpaths for say "/foo", or match any path that happens to contain the word "foo". Thoughts?

let's do the start with then, both would be too much :)

File: anubis-management-api/src/tenants/operations.py [hunk 2]
```diff
@@ -63,10 +63,19 @@ def get_service_paths(db: Session, skip: int = 0, limit: int = 100):
...
         tenant_id: str,
+        name: str = None,
+        startswith: str = None,
         skip: int = 0,
         limit: int = 100):
-    return db.query(models.ServicePath).filter(
-        models.ServicePath.tenant_id == tenant_id).offset(skip).limit(limit).all()
+    db_service_paths = db.query(models.ServicePath).filter(
+        models.ServicePath.tenant_id == tenant_id)
+    if name:
```
Comment (chicco785) [inline anubis-management-api/src/tenants/operations.py:72]: what if both name and startswith? if thing it gets complicated, my idea was to have just the `startswith`:)

Comment (chicco785) [reply anubis-management-api/src/tenants/operations.py:72]: (and use name variable for it would be fine)

**Review by chicco785 — CHANGES_REQUESTED** [review]
what if both name and startswith are used? i think it gets to complicated, let's keep only startswith logic, but using "name" as parameter

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 - Support policy link in response Header
 - Update to opa 0.38.1 and envoy 1.18
+- Support filtering `/tenants/some_tenant/service_paths` by `name` and
```
Comment (chicco785) [inline RELEASE_NOTES.md:9]: only name :)

**Review by chicco785 — CHANGES_REQUESTED** [review]
update release notes consistently with change

**Review by chicco785 — APPROVED** [review]

**Pull request merged**